homework6/counter.py

```diff
@@ -0,0 +1,49 @@
+"""
+Написать декоратор instances_counter, который применяется к любому классу
+и добавляет ему 2 метода:
+get_created_instances - возвращает количество созданых экземпляров класса
+reset_instances_counter - сбросить счетчик экземпляров,
+возвращает значение до сброса
+Имя декоратора и методов не менять
+
+Ниже пример использования
+"""
+
+
+def instances_counter(cls):
+    """
+    Decorate class in order to count a number of reset_instances
+    """
+
+    class Decorated_cl(cls):
+        cls.counter = 0
+
+        def __new__(cls):
+            instance_obj = super(cls, cls).__new__(cls)
+            cls.counter += 1
+            return instance_obj
+
+        @classmethod
+        def get_created_instances(cls):
+            return cls.counter
+
+        @classmethod
+        def reset_instances_counter(cls):
+            reset_instence = cls.counter
+            cls.counter = 0
+            return reset_instence
+
+    return Decorated_cl
```

Comment:
this way you just replace decorated class by this one, so all meta information of initial class would be replaced. please, think about another solution